gsrs-rest-api/src/main/java/gsrs/api/GsrsEntityRestTemplate.java

```diff
@@ -79,26 +79,64 @@ public long count() throws IOException {
     }
     public boolean existsById(I id) throws IOException {
         ResponseEntity<String> response = doGet("("+id + ")", "key",String.class);
-        if(response.getStatusCodeValue() == 404) {
+        if(!response.getStatusCode().is2xxSuccessful()) {
             return false;
         }
         return true;
     }
 
+    // This function in starter is like entityExists but checks to see if ID in json matches the id passed.
+    // I feel like this is kind of good practice, but it might be that I don't understand the upstream code
+    // well enough. For example, exceptions might be raised that prevent the need for this.
+    public Boolean cautiousExistsById(String id, String jsonIdKey) throws IOException {
```

Comment:
This really doesn't feel necessary. The API contract should be to return 404 if the explicit resolve of an ID doesn't work. If there's a time where the API is broken and returns 200 but does not have a valid record (not necessarily the substance), that's a deeper issue that affects a lot of other things. We should just fix that logic if it is a problem.